packages/dev/core/src/Meshes/mesh.ts

```diff
@@ -3395,6 +3403,26 @@ export class Mesh extends AbstractMesh implements IGetSetVerticesData {
                                     uvs.push(currentUVs[2 * facet[j] + k]);
                                 }
                             }
+                            if (currentMatrixIndices !== null && currentMatrixIndices !== void 0) {
+                                for (let k = 0; k < 4; k++) {
+                                    matrixIndices.push(currentMatrixIndices[4 * facet[j] + k]);
+                                }
+                            }
+                            if (currentMatrixWeights !== null && currentMatrixWeights !== void 0) {
+                                for (let k = 0; k < 4; k++) {
+                                    matrixWeights.push(currentMatrixWeights[4 * facet[j] + k]);
+                                }
+                            }
+                            if (currentMatrixIndicesExtra !== null && currentMatrixIndicesExtra !== void 0) {
+                                for (let k = 0; k < 4; k++) {
+                                    matrixIndices.push(currentMatrixIndicesExtra[4 * facet[j] + k]);

```

Comment:
It should be `matrixIndicesExtra` instead of `matrixIndices`.

Reply:
Ho! Good catch!!! I checked the code twice but it always happens

Reply:
Yes, copy/pasting is sneaky ;)